### PR TITLE
Added space between amount and currency

### LINKED ISSUE #5367: formatCurrency is missing space between number and symbol
## User Story

As a user, I want to see the properly formatted amount so that I can be sure about what I'm about to donate.

## Details

In PHP, all currencies that are not explicitly listed get a space between the amount and the symbol:

https://github.com/impress-org/givewp/blob/194015ed4583da3fa84c7fe9be21a156649edd95/includes/currency-functions.php#L309

So when I open a donation form `give-final-total-amount` contains a string like this (because it's rendered by PHP):

> 50,00 CHF

when I click a donation level or manually enter a sum, the display changes to this (because it's rendered by JS):

> 50,00CHF

which is ugly and inconsistent

## Expected Behavior

The amounts should be rendered consistently throughout the site, the JS and PHP renderer should produce the same result

## Steps to Reproduce

1. Go to any form (the prefilled donation amount will be rendered correctly)
2. enter a manual amount or click a donation level
4. the whitespace between the amount and the symbol is now missing

## Visuals

rendered by PHP:

Screenshot_20201013_164942

rendered by Javascript:

Screenshot_20201013_165000

## Acceptance Criteria

-   [ ] Changes to the donation render the currency in the same format as was originally rendered by the server.

### PR DESCRIPTION
**Pull request by karlsauter**
Resolves #5367

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [ ] Changes logged to the `Unreleased` section of `CHANGELOG.md`
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: assets/src/js/plugins/give-api/api.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ const Give = {
 			} else {
 				// Properly position symbol after if selected.
 				if ( 'after' === args.position ) {
-					args.format = '%v%s';
+					args.format = '%v %s';
```
Comment (kjohnson) [inline assets/src/js/plugins/give-api/api.js:109]: The documentation for the [accounting.js#formatMoney](http://openexchangerates.github.io/accounting.js/#methods) library uses the example format of `"%v %s"`, so this looks good to me.

@ravinderk it looks like you initially wrote this, but it was 2 years ago. Do you recall any specific reason to not include the space here, or is this an old typo?

Comment (ravinderk) [reply assets/src/js/plugins/give-api/api.js:109]: @kjohnson I agree with @karlsauter. All currencies do not have the same formatting: https://github.com/impress-org/givewp/issues/5367#issuecomment-736840586. We can add a property to currency config which will tell us whether or not to add space between currency icon/code and amount.

This formatting will break for another currency formatting like `USD`.

Comment (kjohnson) [reply assets/src/js/plugins/give-api/api.js:109]: Ok, so it sounds like we will need to take a different approach.

Thank you, @ravinderk for taking a look.

**Review by kjohnson — CHANGES_REQUESTED** [review]
@karlsauter Perhaps a refactor to use [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) is called for, as you suggested in your precious comment. See https://github.com/impress-org/givewp/pull/5479#discussion_r534278958

And again, we appreciate your contribution and I look forward to working with you on this further.

**Comment by karlsauter** [comment]
@kjohnson got it. I also spotted differences on how the currency is displayed in the backend. I'll upload some pictures later. It seems the preview in the backend is not processed by the same function as the frontend.
Besides that, we should make sure php and JavaScript give the same output. I can take a look at this. Any chance you know where the php function that formats the currency is?

**Comment by kjohnson** [comment]
Closing for now, but happy to revisit this in the future.

**Pull request closed without merging**